lgc/state/LgcContext.cpp

```diff
@@ -303,3 +303,21 @@ PassManagerCache *LgcContext::getPassManagerCache() {
     m_passManagerCache = new PassManagerCache(this);
   return m_passManagerCache;
 }
+
+// =====================================================================================================================
+// Get the size in bytes of a resource (image) descriptor.
+unsigned LgcContext::getDescriptorSizeResource() const {
+  return getTargetInfo().getGpuProperty().descriptorSizeResource;
+}
+
+// =====================================================================================================================
+// Get the size in bytes of a sampler descriptor.
+unsigned LgcContext::getDescriptorSizeSampler() const {
+  return getTargetInfo().getGpuProperty().descriptorSizeSampler;
+}
+
+// =====================================================================================================================
+// Get the size in bytes of a buffer descriptor.
+unsigned LgcContext::getDescriptorSizeBuffer() const {
+  return getTargetInfo().getGpuProperty().descriptorSizeBuffer;
+}
```

Comment:
This smells of the kind of pattern that I feel hurts us via death by a thousand cuts.

These numbers are always the same. Indirecting them in this way isn't needed now, and slows down both the software and the developers:

1. It slows down the software because what could have been an inlined constant, percolating up to additional optimizations, at least in an LTO build, now becomes a load from memory that can't be particularly well optimized. Sure, it's not a big issue in isolation, but we have this kind of thing **everywhere** (also in the LLVM backend, where the situation is arguably worse).

2. Knowing that descriptors are always the same size is valuable for an understanding of the overall system. Somebody who doesn't know much about our hardware (yet?) will not be able to learn this fact from looking at this code, unless they dig further.

Please consider just making those global constants instead and removing them from `lgc::GpuProperty` entirely. On the rather unlikely chance that these numbers ever become dynamic, we're going to have to revisit all relevant code anyway -- it's not like changing the return value of those functions is just magically going to work.

Reply:
Resolved in #787, that this PR is now rebased on.